### DISCUSSION TITLE
Error : contrasts can be applied only to factors with 2 or more levels

### DISCUSSION THREAD

**Discussion by yuliristiana** [teamalgoritma/community#765] [Q&A] [answered]
Cara menyelesaikan error seperti itu bagaimana? ada yang tau dan punya solusinya kah?

![image](https://github.com/teamalgoritma/community/assets/134671802/c14be8e8-4b2c-4fb6-9107-409f29a58836)

**Comment by kinantytasyao** [comment]
Sepertinya Ibu memiliki minimal satu dari variabel/kolom prediktor yang hanya memiliki satu kategori/nilai (konstan), sehingga jika benar hal tersebut terjadi solusinya adalah dengan membuang kolom dengan nilai konstan tersebut. Namun, jika tidak didapati kolom dengan nilai konstan, silakan lampirkan head(PCOS_train_up) ya Bu agar dapat kami tracing lebih lanjut...

Reference:
- https://www.statology.org/contrasts-applied-to-factors-with-2-or-more-levels/
- https://stackoverflow.com/questions/44200195/how-to-debug-contrasts-can-be-applied-only-to-factors-with-2-or-more-levels-er

Demikian yang dapat kami sampaikan. Semoga membantu!

**Reply by yuliristiana** [reply]
Saya sudah coba debug kebetulan dengan link yang sama, tapi error nya masih ada dan belum mendapatkan solusinya. 

'Aneh' nya jika data fungsi _glm_ membutuhkan data factor dengan min 2 level atau lebih maka seharusnya data di kolom PCOS sudah memenuhi kriteria _contrasts_ di fungsi _glm_ karena ada 2 data unique yaitu 'yes' & 'no' .

ini saya lampirkan _head(PCOS_train_up_) yg mungkin bisa membantu mba Kinan untuk tracing lebih lanjut. Terima kasih!
![image](https://github.com/teamalgoritma/community/assets/134671802/c52b728e-f9a4-4c3f-8404-f81d75b04608)

**Reply by kinantytasyao** [reply]
Bukan kolom target PCOS nya Bu, tetapi salah satu dari 36 kolom yang digunakan sebagai prediktor. Karena head dari datanya tidak menunjukkan semua kondisi kolomnya, boleh tolong untuk dilampirkan saja csv data nya Bu atau bisa ditunjukkan pakai str() untuk menampilkan semua kolom?

**Reply by yuliristiana** [reply]
Hai mba Kinan,

Berikut mba file yang saya gunakan. File ini sama dengan file yang saya gunakan pada pertanyaan diatas.

[PCOS_data_train.csv](https://github.com/teamalgoritma/community/files/12575571/PCOS_data_train.csv)

**Reply by kinantytasyao** [reply, marked as answer]
Baik, setelah saya cek data csv dan head(data) yang Ibu lampirkan sepertinya Ibu belum melakukan pengubahan data type. Bisa dilihat pada head(data) variabel "Height" masih bertipe data karakter. Seharusnya ketika melakukan pemodelan regresi logistik, data yang digunakan sudah bersih yaitu tipe datanya sudah benar dan tidak ada missing values untuk setiap kolom. Kami sangat menganjurkan Ibu melakukan pengubahan tipe data yg benar untuk 44 kolom variabel yang terdapat pada data, pengubahan tipe data ini bisa disesuaikan dengan membaca deskripsi masing-masing variabel pada dokumentasi data.

**Comment by ramdhinafinita** [comment]
Halo Ibu @yuliristiana, untuk menjaga agar forum diskusi berjalan kondusif, kami izin melakukan close pada diskusi ini karena belum ada respon setelah 7 hari respon terakhir dari kami. Semoga dapat dipahami, terimakasih.
